pkg/kubelet/server/server.go

```diff
@@ -216,6 +211,12 @@ func NewServer(
 		auth:                       auth,
 		restfulCont:                &filteringContainer{Container: restful.NewContainer()},
 		redirectContainerStreaming: redirectContainerStreaming,
+		rt: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				// TODO: authenticate connections to proxied streaming servers.
+				InsecureSkipVerify: true,
```

Comment:
That's a pretty big TODO. Will it happen in the next PR or undefined time in the future?

Reply:
Before this change, we connected over HTTP so this is an improvement. I have some ideas on how to do this but my primary motivation is to get the entire flow of streaming fro kubectl -> apiserver -> kubelet -> container runtime over TLS so that it support HTTP2. This is the last arrow and at least it's encrypted and integrity checked now.

Reply:
Hmm, what in this change forces it to go over HTTPS over HTTP?

Reply:
TLSOptions.TLSConfig doesn't have any certs in it when using self signed or certs from disk:

https://github.com/kubernetes/kubernetes/blob/2b0212de9cdf4c2cc8d5619dcc26c1c0b8917203/cmd/kubelet/app/server.go#L840

Only the cert/key paths are provided. Streaming server passes an empty config to the http server:

https://github.com/kubernetes/kubernetes/blob/2b0212de9cdf4c2cc8d5619dcc26c1c0b8917203/pkg/kubelet/server/streaming/server.go#L149
https://github.com/kubernetes/kubernetes/blob/2b0212de9cdf4c2cc8d5619dcc26c1c0b8917203/pkg/kubelet/server/streaming/server.go#L245

Reply:
After this change, TLSOptions is eliminated.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/server.go#L91-L95

So there's no confusion on whether to use the TLSConfig or CertFile, KeyFile.

Now the streaming server always starts up with kubelet's serving cert.